arbnode/inbox_tracker.go

```diff
@@ -343,15 +360,23 @@ func (t *InboxTracker) AddDelayedMessages(messages []*DelayedInboxMessage, hardR
 		}
 		nextAcc = message.AfterInboxAcc()
 
-		msgKey := dbKey(rlpDelayedMessagePrefix, seqNum)
+		delayedMsgKey := dbKey(rlpDelayedMessagePrefix, seqNum)
 
 		msgData, err := rlp.EncodeToBytes(message.Message)
 		if err != nil {
 			return err
 		}
 		data := nextAcc.Bytes()
 		data = append(data, msgData...)
-		err = batch.Put(msgKey, data)
+		err = batch.Put(delayedMsgKey, data)
+		if err != nil {
+			return err
+		}
+
+		parentChainBlockNumberKey := dbKey(parentChainBlockNumberPrefix, seqNum)
+		parentChainBlockNumberByte := make([]byte, 8)
+		binary.BigEndian.PutUint64(parentChainBlockNumberByte, message.ParentChainBlockNumber)
+		err = batch.Put(parentChainBlockNumberKey, parentChainBlockNumberByte)
```

Comment:
To save space in the db for L2s, we should only store this if `message.ParentChainBlockNumber != message.Message.Header.BlockNumber`. Depending on if message pruning support or this PR gets merged first, we should also add support for pruning in the latter PR to get merged (as-is it's tricky because they're mutually dependent so this can wait).